flow/TLSConfig.actor.cpp

```diff
@@ -173,14 +173,10 @@ ACTOR static Future<Void> readEntireFile( std::string filename, std::string* des
 	state Reference<IAsyncFile> file = wait(IAsyncFileSystem::filesystem()->open(filename, IAsyncFile::OPEN_READONLY | IAsyncFile::OPEN_UNCACHED, 0));
 	state int64_t filesize = wait(file->size());
 	if (filesize > FLOW_KNOBS->CERT_FILE_MAX_SIZE) {
-		throw tls_error();
+		throw file_too_large();
 	}
 	destination->resize(filesize);
-	int rc = wait(file->read(const_cast<char*>(destination->c_str()), filesize, 0));
-	if (rc != filesize) {
-		// File modified during read, probably.  The mtime should change, and thus we'll be called again.
-		throw tls_error();
-	}
+	wait(file->read(const_cast<char*>(destination->c_str()), filesize, 0));
```

Comment:
This is the change that broke the build, since the future is of type int. You could change it to use `success()` if we don't care about the size anymore:

```suggestion
	wait(success(file->read(const_cast<char*>(destination->c_str()), filesize, 0)));
```

Reply:
It turns out checking the size gets us nothing.  If the read comes back as -1, the AsyncFile classes all translate that into io_error() already.  If the read comes back the correct size, the file might have been larger on disk, and thus our read might still be corrupt.  Even if it's smaller, it might still be a valid cert, so we might as well try.